### PR TITLE
XW-3296 | new dropdowns

### PR DESCRIPTION
**Pull request by latata**
https://wikia-inc.atlassian.net/browse/XW-3296

@Wikia/x-wing

### CODE REVIEW

File: style-guide/styles/wds-components/_tabs.scss [hunk 1]
```diff
@@ -11,69 +11,40 @@
 	$tab-default-color: $wds-color-link;
 	$tab-hover-color: darken($tab-default-color, 20%);
 	$tab-icon-size: 14px;
-	$tab-chevron-size: 10px;
+	$tab-chevron-size: $wds-icon-tiny-size;
 	$tab-icon-margin: 6px;
 
 	cursor: pointer;
 	line-height: 14px;
```
Comment (rybmat) [inline style-guide/styles/wds-components/_tabs.scss]: $wds-typescale-size-minus-1 ?

File: dist/scss/wds-components/_dropdowns.scss [hunk 2]
```diff
@@ -1,5 +1,9 @@
+$border-radius: 4px;
+$dropdown-border-width: 1px;
```
Comment (weronika) [inline dist/scss/wds-components/_dropdowns.scss]: I guess these variables should be internal. Or if not - they should have 'wds-' prefix.

File: dist/scss/wds-components/_dropdowns.scss [hunk 3]
```diff
@@ -17,30 +21,188 @@
...
+			border: 0;
+		}
+	}
+}
+
+.wds-dropdown__content .wds-list.wds-is-linked .wds-dropdown-level-2 {
+	margin-right: 0;
+
+	> a {
+		padding-right: 18px;
```
Comment (weronika) [inline dist/scss/wds-components/_dropdowns.scss]: maybe we could use some calculation of existing dropdown padding/margin variables? I guess they are somehow correlated.

File: style-guide/styles/wds-components/_tabs.scss (hunk 1 above)
Comment (latata) [inline style-guide/styles/wds-components/_tabs.scss]: `$wds-typescale` variables are not used for `line-height`

File: style-guide/styles/wds-components/_dropdowns.scss [hunk 4]
```diff
@@ -17,30 +18,189 @@
 
 	&__content {
 		background-color: $wds-color-white;
-		box-shadow: 0 0 10px 0 rgba($wds-color-black, .2);
+		border: $wds-dropdown-border-width solid $wds-color-faint-gray;
+		border-radius: $wds-dropdown-border-radius;
 		display: none;
-		left: 0;
+		left: 50%;
+		min-width: 60%;
```
Comment (rogatty) [inline style-guide/styles/wds-components/_dropdowns.scss]: Let's add a comment explaining what this is for.

File: app/templates/route-components/buttons.hbs [hunk 5]
```diff
@@ -317,16 +331,19 @@ html[lang^='zh'] &__start-a-wiki-button {
 				{{wds-icon name='pencil' size='small'}}
 				Edit
 			{{/wds-button}}
-			{{#wds-button class='wds-is-squished wds-dropdown wds-is-active'}}
-				{{wds-dropdown-toggle}}
+			<div class="wds-dropdown">
```
Comment (hakubo) [inline app/templates/route-components/buttons.hbs]: for consistency should we use {{wds-dropdown}} here?

File: dist/scss/wds-components/_dropdowns.scss [hunk 6]
```diff
@@ -17,30 +16,190 @@
...
+						@include wds-dropdown-linked-list-item-hover;
+					}
+				}
+			}
 		}
 	}
 
-	&.wds-is-active .wds-dropdown {
-		&__content {
+	&::after {
```
Comment (hakubo) [inline dist/scss/wds-components/_dropdowns.scss]: What do you think about?
&::after, &::before {
//'common' properties

&::hover {
display: block;
}
}

&::before {
custom before prop
}

&::after {
}

File: dist/scss/wds-components/_dropdowns.scss [hunk 7]
```diff
@@ -17,30 +16,190 @@
...
+	}
+
+	&.wds-has-shadow, &.wds-has-dark-shadow {
+		.wds-dropdown__content {
+			border: 0;
+		}
+	}
+}
+
+.wds-dropdown__content .wds-list.wds-is-linked .wds-dropdown-level-2 {
```
Comment (hakubo) [inline dist/scss/wds-components/_dropdowns.scss]: is it really necessary to have it that long? shouldn't .wds-dropdown-level-2 be enough?

File: dist/scss/wds-components/_notifications.scss [hunk 8]
```diff
@@ -47,15 +47,22 @@
 		max-width: $content-padding * 2 + 320px;
 		min-height: 80px;
 		min-width: $content-padding * 2 + 320px;
-		white-space: normal;
 		width: 100%;
+
+		// that's here because we need to override DS dropdowns list styles
```
Comment (hakubo) [inline dist/scss/wds-components/_notifications.scss]: imagine it's 2018 - would you understand that comment?
why do we need to override it?

**Comment by rogatty** [comment]
The Upstream team has the changes ready in https://github.com/Wikia/upstream/pull/1362 and is waiting for the release. Let's merge it.

**Pull request merged**